config/crd/bases/ols.openshift.io_configs.yaml

```diff
@@ -0,0 +1,196 @@
+                          type: object
+                          x-kubernetes-map-type: atomic
+                        models:
+                          description: List of models from the provider
+                          items:
+                            description: ModelSpec defines the desired state of in-memory
+                              cache.
+                            properties:
+                              name:
+                                description: Model name
+                                type: string
+                              url:
+                                description: Model API URL
+                                type: string
+                            required:
+                            - name
+                            type: object
+                          type: array
+                        name:
+                          description: Provider name
+                          type: string
+                        url:
+                          description: Provider API URL
+                          type: string
+                      type: object
+                    type: array
+                required:
+                - providers
+                type: object
+              ols:
+                description: OLSSpec defines the desired state of OLS deployment.
+                properties:
+                  classifierModel:
+                    description: Classifier model name
+                    type: string
+                  classifierProvider:
+                    description: Classifier provider name
+                    type: string
+                  conversationCache:
+                    description: Conversation cache settings
+                    properties:
+                      memory:
+                        description: MemorySpec defines the desired of in-memory cache.
+                        properties:
+                          maxEntries:
+                            default: 1000
+                            description: 'Maximum number of cache entries. Default:
+                              "1000"'
+                            type: integer
+                        type: object
+                      redis:
+                        description: RedisSpec defines the desired state of Redis.
+                        properties:
+                          host:
+                            description: Redis host
+                            type: string
+                          maxMemory:
+                            anyOf:
+                            - type: integer
+                            - type: string
+                            description: Redis maxmemory
+                            x-kubernetes-int-or-string: true
+                          maxMemoryPolicy:
+                            default: allkeys-lru
+                            description: 'Redis maxmemory policy. Default: "allkeys-lru"'
```

Comment:
I dont understand this use case clear enough. 1. Are we planning to use the REDIS instance that user is running , if then I dont see the PASSWORDS/CERTS to connect. 2. If we are planning to host it in our deployment we dont have stories . Lost here @bparees 
Also should we have configmap for the redis parameters , so that we dont keep updating to the needs ?

Reply:
good questions @xrajesh 

i think we're probably premature on the redis config, in general (we don't even know for sure that we're using redis for the cache, though it seems likely).

but whatever we have for the "real" cache impl:
1) it will run in a separate pod because we need multiple ols pod instances talking to a shared/common cache instance (the cache itself may also run multiple pods for scaling purposes, but expose a single cache service endpoint)
2) so it will need a username/password so OLS can talk to it but that is something the operator will need to wire up, not something the admin will provide in the ols configuration CR.
3) no we should not use a configmap for redis parameters.  configmap content is notoriously difficult to version and validate for correctness.  So to the extent we want to allow the admin to configure the redis instance, we should expose those configuration parameters in the ols config CRD and then the operator will need to wire those values through to redis (now it may do that by generating a config file in a configmap and mounting it into the redis deployment, but that configmap would not be edited by the user).  Again, a bit premature to think about redis configuration right now.